pkg/util/array.go

```diff
@@ -1,14 +1,14 @@
 package util
 
 // ArrayReverse return a reversed version of the given byte slice.
-func ArrayReverse(b []byte) []byte {
+func ArrayReverse(arr []byte) []byte {
 	// Protect from big.Ints that have 1 len bytes.
-	if len(b) < 2 {
-		return b
+	if len(arr) < 2 {
+		return arr
 	}
-	dest := make([]byte, len(b))
-	for i, j := 0, len(b)-1; i < j; i, j = i+1, j-1 {
-		dest[i], dest[j] = b[j], b[i]
+	for i := len(arr)/2 - 1; i >= 0; i-- {
+		opp := len(arr) - 1 - i
+		arr[i], arr[opp] = arr[opp], arr[i]
 	}
-	return dest
+	return arr
 }
```

Comment:
func TestArrayReverseNastya(t *testing.T) {
	data, err := hex.DecodeString("0bcd2978634d961c24f5aea0802297ff128724d6")
	if err != nil {
		t.Fatal(err)
	}
	have := ArrayReverse(data)
	want := []byte{214, 36, 135, 18, 255, 151, 34, 128, 160, 174, 245, 36, 28, 150, 77, 99, 120, 41, 205, 11}
	if bytes.Compare(have, want) != 0 {
		t.Fatalf("expected %v got %v", want, have)
	}
	have2 := ar(data)
	if bytes.Compare(have2, want) != 0 {
		t.Fatalf("expected %v got %v", want, have)
	}

	if bytes.Compare(have2, have) != 0 {
		t.Fatalf("expected %v got %v", want, have)
	}
}

Reply:
both results are the same.. not quite clear why to change..